File: BMICalculator/app/src/main/java/com/example/bmicalculator/MainActivity.java
package com.example.bmicalculator;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
        EditText weight, height;
        TextView resulttext;
        String calculation, BMIresult;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        weight = findViewById(R.id.weight);
        height = findViewById(R.id.height);
        resulttext = findViewById(R.id.result);

    }

    public void calculateBMI(View view) {
        String w = weight.getText().toString();
        String h = height.getText().toString();

        float weightValue = Float.parseFloat(w);
        float heightValue = Float.parseFloat(h)/100;

        float bmi = (weightValue / (heightValue * heightValue ));

        if(bmi<18.5) {
            BMIresult = "Underweight";
        }
        else if(bmi<25){
            BMIresult = "Healthy Weight";
        }
        else if(bmi<30){
            BMIresult = "Overweight";
        }
        else if(bmi>30){
            BMIresult = "Obese";
        }

        calculation = "Result:\n" +bmi+ "\n" +BMIresult;

        resulttext.setText(calculation);
    }
}

File: BMIKotlin/BMICalculator/app/src/main/java/com/example/bmicalculator/MainActivity.kt
package com.example.bmicalculator

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
   // private val SPLASH_TIME_OUT:Long = 3000 // 1 sec

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


            val homeIntent = Intent(this@MainActivity, HomeActivity::class.java)
            startActivity(homeIntent)
            finish()

    }

}